master/internal/api_trials_intg_test.go

```diff
@@ -467,6 +467,40 @@ func TestTrialsNonNumericMetrics(t *testing.T) {
 	})
 }
 
+func TestUnusualMetricNames(t *testing.T) {
+	api, curUser, ctx := setupAPITest(t, nil)
+	expectedMetricsMap := map[string]any{
+		"a.loss": 1.5,
```

Comment:
Maybe exhaustively test ascii printable characters here to make sure no other characters cause a similar issue